K2Bridge/Visitors/Aggregations/AggregationDictionaryVisitor.cs

```diff
@@ -70,6 +71,47 @@ public string BuildBucketAggregationQuery(BucketAggregation bucketAggregation, B
             query.Append($"{KustoQLOperators.CommandSeparator} {KustoQLOperators.Summarize} {bucketAggregation.SubAggregationsKustoQL}");
             query.Append($"{definition.BucketExpression};");
 
+            // Create a table named "metadata", with one column per aggregation and dynamic value.
+            if (definition.Metadata != null)
+            {
+                query.Append(BuildMetadataQuery(definition.Metadata));
+            }
+
+            return query.ToString();
+        }
+
+        /// <summary>
+        /// Given a metadata dictionary, generates a Kusto QL statement creating the metadata table:
+        /// datatable(['2']:string) [dynamic(['val1','val2', 'val3'])] | as metadata;
+        /// </summary>
+        /// <param name="metadata">A metadata dictionary.</param>
+        /// <returns>A string containing the Kusto QL statement.</returns>
+        public string BuildMetadataQuery(Dictionary<string, string> metadata)
```

Comment:
Thought - instead of building one dynamic cell that you then parse as json,
why not have a row in the table for each value? seems less "hacky"

Reply:
Right, I was trying to avoid "querying" the table, i.e. find all the rows where the key matches and concatenate the values. But maybe that is not that complicated. I will give it a try.

Reply:
I have changed the schema to a more traditional key/value scheme. LMKWYT

Reply:
It looks good to me. This is more performant (but probably neglegable since they will probably not be more than a handful of filters/ranges), but I think it's more readable/understandable. Do you agree or do you still prefer the dynamic way?

Reply:
I am happy with the new table approach, the dynamic column was indeed a bit obscure!